testing_app/test_driver/perf_test.dart

```diff
@@ -0,0 +1,123 @@
+        // Scroll until the item to be found appears.
+        // Use dxScroll to scroll horizontally and dyScroll
+        // to scroll vertically.
+        await driver.scrollUntilVisible(
+          listFinder,
+          lastItem,
+          dyScroll: -1000.0,
+        );
+
+        // Check if the item contains the correct text.
+        expect(await driver.getText(lastItem), 'Item 49');
+
+        // Scroll back up all the way.
+        await driver.scrollUntilVisible(
+          listFinder,
+          firstItem,
+          dyScroll: 1000.0,
+        );
+
+        // Check if the item contains the correct text.
+        expect(await driver.getText(firstItem), 'Item 0');
+      });
+
+      // Convert the Timeline into a TimelineSummary that's easier to
+      // read and understand.
+      final scrollingSummary = TimelineSummary.summarize(scrollingTimeline);
+
+      // Then, save the summary to disk.
+      await scrollingSummary.writeSummaryToFile('scrolling', pretty: true);
+
+      // Optionally, write the entire timeline to disk in a json format.
+      // This file can be opened in the Chrome browser's tracing tools
+      // found by navigating to chrome://tracing.
+      await scrollingSummary.writeTimelineToFile('scrolling', pretty: true);
+    });
+
+    test('Favorites operations test', () async {
+      // Record a performance profile as operations are performed
+      // on the favorites list.
+      final operationsTimeline = await driver.traceAction(() async {
+        // Create a list of icon keys.
+        final iconKeys = [
+          'icon_0',
+          'icon_1',
+          'icon_2',
+        ];
+
+        // Add first three items to favorites.
+        for (var icon in iconKeys) {
+          // Tap onto the icon.
+          await driver.tap(find.byValueKey(icon));
+
+          // Verify if appropriate message appears.
+          await driver.waitFor(find.text('Added to favorites.'));
+        }
+
+        // Tap onto the favorites button on the AppBar.
+        // The Favorites List should appear.
+        await driver.tap(find.text('Favorites'));
+
+        final removeIconKeys = [
+          'remove_icon_0',
+          'remove_icon_1',
+          'remove_icon_2',
+        ];
+
+        // Remove all the items from favorites.
+        for (final iconKey in removeIconKeys) {
+          // Tap onto the remove icon.
+          await driver.tap(find.byValueKey(iconKey));
+
+          // Verify if appropriate message appears.
+          await driver.waitFor(find.text('Removed from favorites.'));
+        }
+      });
+
+      // Convert the Timeline into a TimelineSummary.
+      final operationsSummary = TimelineSummary.summarize(operationsTimeline);
+
+      // Then, save the summary to disk.
+      await operationsSummary.writeSummaryToFile('favorites_operations',
+          pretty: true);
+
+      // Optionally, write the entire timeline to disk in a json format.
```

Comment:
nit: delete the word Optionally, it isn't required =)

Reply:
Removed 👍